cmd/osm-controller/validate.go

```diff
@@ -32,42 +34,46 @@ func validateCLIParams() error {
 		return errors.Errorf("Invalid --webhook-config-name value: '%s'", webhookConfigName)
 	}
 
+	if caBundleSecretName == "" {
+		return errors.Errorf("Please specify the CA bundle secret name using --ca-bundle-secret-name")
+	}
+
 	return nil
 }
 
 func validateCertificateManagerOptions() error {
-	switch *osmCertificateManagerKind {
-	case tresorKind:
+	switch osmCertificateManagerKind {
+	case certproviderutil.TresorKind:
 		break
-	case vaultKind:
+	case certproviderutil.VaultKind:
 		if err := validateVaultParams(); err != nil {
 			return err
 		}
-	case certmanagerKind:
+	case certproviderutil.CertManagerKind:
 		if err := validateCertManagerParams(); err != nil {
 			return err
 		}
 	default:
 		return errors.Errorf("Invalid certificate manager kind %s. Please specify a valid certificate manager[%v] \n",
-			*osmCertificateManagerKind, strings.Join(validCertificateManagerOptions, "|"))
+			osmCertificateManagerKind, strings.Join(certproviderutil.ValidCertificateProviders, "|"))
 	}
 
 	return nil
 }
 
 func validateCertManagerParams() error {
 	if len(caBundleSecretName) == 0 {
-		return errors.Errorf("Please specify --%s as the Secret name containing the cert-manager CA at 'ca.crt'", caBundleSecretNameCLIParam)
+		return errors.Errorf("Please specify --ca-bundle-secret-name as the Secret name containing the cert-manager CA at 'ca.crt'")
```

Comment:
Isn't it better to keep `--ca-bundle-secret-name` as a string constant so next time we have to change it it is in one place?

Reply:
I understand why it was done this way, but this is true for all cmd line params, and creating constants for each of these seems to be a bit unnecessary. Assuming these option names rarely change, maybe a simple search and replace could suffice. I only did this to be consistent with other options, not other reason.